### PR TITLE
Add downstream libunwind changes from emscripten

### PR DESCRIPTION
**Pull request by aheejin**

### CODE REVIEW

**Comment by sbc100** [comment]
Did you use `system/lib/update_libcxx.py` to generate this change?

Can you update the version in `system/lib/libcxx/readme.txt`?

**Comment by aheejin** [comment]
I removed `XXX EMSCRIPTEN` from libunwind src, because it looks we don't use it anymore in other libraries.

> Did you use `system/lib/update_libcxx.py` to generate this change?

No, it was a single line, so I did it manually.

> Can you update the version in `system/lib/libcxx/readme.txt`?

This repo doesn't have it. I'll update it in https://github.com/emscripten-core/emscripten/pull/20088 along with other libraries' `readme.txt`. (Other libraries don't have any changes between 16.0.4 and 16.0.6 other than some cmake files, which we don't maintain.)

**Comment by sbc100** [comment]
Oh wait, sorry I was confused.  This is a change to our fork of llvm.

To update the `emscripten-libs-16` branch to 16.0.6 you would do `git merge llvmorg-16.0.6`  and then upload the merge commit as a PR.  

Then I imagine this PR would be called something like "Add downstream libunwind changes from emscripten"

**Comment by aheejin** [comment]
> To update the `emscripten-libs-16` branch to 16.0.6 you would do `git merge llvmorg-16.0.6` and then upload the merge commit as a PR.

Done in #3.

File: libunwind/include/unwind_itanium.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ struct _Unwind_Exception {
                             _Unwind_Exception *exc);
 #if defined(__SEH__) && !defined(__USING_SJLJ_EXCEPTIONS__)
   uintptr_t private_[6];
-#else
+#elif !defined(__USING_WASM_EXCEPTIONS__)
```
Comment (sbc100) [inline libunwind/include/unwind_itanium.h:27]: Is this a macro that already exists somewhere in libunwind ?

**Review by sbc100 — APPROVED** [review]

File: libunwind/include/unwind_itanium.h (hunk 1 above)
Comment (aheejin) [reply libunwind/include/unwind_itanium.h:27]: No we add it to the command line both for [libcxxabi](https://github.com/emscripten-core/emscripten/blob/e1236625707595d253c189c5b152ed809afc54a5/tools/system_libs.py#L1506) and [libunwind](https://github.com/emscripten-core/emscripten/blob/e1236625707595d253c189c5b152ed809afc54a5/tools/system_libs.py#L1606).

**Pull request merged**